packages/calcite-components/src/components/shell/shell.tsx

```diff
@@ -131,6 +137,24 @@ export class Shell implements ConditionalSlotComponent {
     });
   };
 
+  handlePanelTopChange = (event: Event): void => {
+    const panelTop = event.target as HTMLSlotElement;
+    this.hasPanelTop = !!panelTop;
+
+    this.updateHasOnlyPanelBottom();
+  };
+
+  handlePanelBottomChange = (event: Event): void => {
+    const panelBottom = event.target as HTMLSlotElement;
+    this.hasPanelBottom = !!panelBottom;
+
+    this.updateHasOnlyPanelBottom();
+  };
+
+  updateHasOnlyPanelBottom = (): void => {
```

Comment:
This can be removed if we use a watch to watch the two states we're interested in to update the other state.